packages/web3/src/index.ts

```diff
@@ -323,4 +323,5 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
  */
 import Web3 from './web3';
 
-export = Web3;
+export * from './web3';
```

Comment:
Those changes revert back the modifications made at: https://github.com/web3/web3.js/pull/5221/files#diff-bef53270c59945a684e63b962974a96318e07549c207ea92bbc9c369e35bc476R326
And it will cause backward incompatibility with version 1.x, right?

Reply:
It doesn't break anything as mentioned [here](https://github.com/web3/web3.js/pull/5393#discussion_r960143755)

Reply:
Yes, on line `327`, `Web3` is still the `default` export of the package:

```typescript
export default Web3;
```

An example of this working is in the [contract_method_wrappers tests](https://github.com/web3/web3.js/blob/6093e5bb5aee9fd87f00c5fbfcbe35f3b5d1ddd0/tools/web3-plugin-example/test/unit/contract_method_wrappers.test.ts#L17)

Reply:
I created a simple javascript project with the following code:
```javascript
const Web3 = require('web3'); 

const web3 = new Web3();
```

And only when using this branch, the following error occurs:
> TypeError: Web3 is not a constructor

So, could you please revert back any changes made to how Web3 is exported? Or else, if it is necessary to have those changes for the plugin, then please let me know where exactly this is needed so we can investigate more on this matter...

Reply:
Interestingly, it appears this still works if you do

```typescript
const Web3 = require('web3');
const web3 = new Web3.default('http://127.0.0.1:8545');
```

Reply:
The above give us the following eslint error though

```bash
web3.js/packages/web3/src/index.ts
  324:1  error    Cannot use import declarations in modules that export using CommonJS (module.exports = 'foo' or exports.bar = 'hi')  import/no-import-module-exports
```

Reply:
Do you face any problems reverting this MR's changes to the Web3 export statement?
It was `export = Web3;` not `module.exports = Web3;`
It was:
```typescript
import Web3 from './web3';

export = Web3;
```

Reply:
Yes, TypeScript module augmentation seems to be designed to work with named ES modules. So only exporting `Web3` as a CommonJS module results in the following error

```bash
test/unit/contract_method_wrappers.test.ts:27:16 - error TS2664: Invalid module name in augmentation, module 'web3' cannot be found.

    27 declare module 'web3' {
```

I've tried many different ways to work around this, but it boils down to us not being able to mix CommonJS and ES modules, the latter being needed to provide module augmentation

However, we can use

```typescript
import Web3 from './web3';

export * from './web3';
export default Web3;
```

and the user would still be able to do

```typescript
const Web3 = require('web3').default;
```

but of course this still results in a breaking change and module augmentation doesn't work

Reply:
@spacesailor24 Limitation is for module augmentation there must be named export as mentioned in ts docs Module Augmentation part: 
"<img width="840" alt="image" src="https://user-images.githubusercontent.com/86780488/196973068-a43a160d-a9bd-4e39-a174-6d5af1b03051.png">"


so If we do named export of Web3 in current CJS settings module augmentation works fine.

As discussed earlier for supporting legacy applications, and not introducing breaking change, I am in favour of keeping current Web3 4x export settings ( Also mentioned by [Muhammad-Altabba](https://github.com/Muhammad-Altabba) ) and there is a workaround for module augmentation for now:

so in user plugin project where web3 augmentation is required: if Web3 is imported and reexported as named, users can augment. 

**web3.ts** in user project
```
import Web3 from 'web3';
export { Web3 };
```

and **pluginConsumer.ts** using named reexported web3 from user's web3.ts
```
import { Web3 } from './web3';
import { Web3Context } from 'web3-core';
import { EthExecutionAPI } from 'web3-types';

import { ContractMethodWrappersPlugin } from './contract_method_wrappers';
import { ERC20TokenAbi } from './ERC20Token';

declare module './web3' {
    interface Web3 extends Web3Context<EthExecutionAPI> {
        contractMethodWrappersPlugin: ContractMethodWrappersPlugin;
    }
}
```

Reply:
@Muhammad-Altabba @jdevcs [This](https://github.com/web3/web3.js/pull/5393/commits/56bf6b7e00bc3f0e21a5aed30525a88c285b46bf) commit adds the work around mentioned above by Junaid

Reply:
Thanks @jdevcs  for the good workaround.
And thanks @spacesailor24 for the good implementation.
I think the MR is ready to be merged after updating the CHANGELOG.md.